header-rewrite-filter/header_processor.cc

```diff
@@ -224,17 +224,22 @@ namespace HeaderRewriteFilter {
 
             const Utility::MatchType match_type = Utility::StringToMatchType(*(start + 3));
 
-            std::string string_to_compare;
+            std::string string_to_compare = std::string(*(start + 4));
 
             switch (match_type) {
                 case Utility::MatchType::Exact:
-                    string_to_compare = std::string(*(start + 4));
-                    source_ = *(start + 1);
+                    source_ = std::string(*(start + 1));
                     matcher_ = [string_to_compare](std::string source) { return (source.compare(string_to_compare) == 0); };
                     break;
-                // TODO: implement this
+                case Utility::MatchType::Prefix:
+                    source_ = std::string(*(start + 1));
```

Comment:
Just checking -- this `source_` is later used when calling `matcher_`. So as-is, we could just evaluate `matcher_(source_)` right here at parse-time. But that isn't very useful, we really want to be able to match on variables at runtime. I'm guessing the plan is to evaluate `source_` as a variable later, so we can match things like metadata, header values, and variables at request-time? (or maybe it already does that and I'm just not seeing it?)

Reply:
Nevermind, I see the answer is yes in https://github.com/DataDog/envoy-filter-example/pull/26